### PR TITLE
Use consul as an external pillar source. Requires python-consul

### PR DESCRIPTION
**Pull request by devopsbrett**

### CODE REVIEW

**Comment by jtand** [comment]
Go Go Jenkins!

**Comment by jfindlay** [comment]
Go Go Jenkins!

File: README.rst [hunk 1]
```diff
@@ -66,3 +66,4 @@ Reporting Issues
 To report an issue with Salt, please follow the guidelines for filing bug reports:
 
 * http://docs.saltstack.com/en/latest/topics/development/reporting_bugs.html
+ping
```
Comment (bechtoldt) [inline README.rst]: You can trigger a Jenkins build by posting "Go Go Jenkins!" as a comment. See @jtand 's comment.

File: salt/pillar/consul_pillar.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    keyvalue = keys.pop()
+    pil = {keyvalue: pillar_value}
+    keys.reverse()
+    for k in keys:
+        pil = {k: pil}
+
+    return dict_merge(ret, pil)
+
+
+def dict_merge(d1, d2):
```
Comment (bechtoldt) [inline salt/pillar/consul_pillar.py]: can you make use of https://github.com/saltstack/salt/blob/develop/salt/utils/dictupdate.py?

File: salt/pillar/consul_pillar.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        return ret
+    array_data = value.split('\n')
+    pillar_value = array_data[0] if len(array_data) == 1 else array_data
+    keyvalue = keys.pop()
+    pil = {keyvalue: pillar_value}
+    keys.reverse()
+    for k in keys:
+        pil = {k: pil}
+
+    return dict_merge(ret, pil)
```
Comment (bechtoldt) [inline salt/pillar/consul_pillar.py]: can you make use of https://github.com/saltstack/salt/blob/develop/salt/utils/dictupdate.py?

File: README.rst (hunk 1 above)
Comment (jfindlay) [inline README.rst]: Actually only salt admins can retrigger tests using this method, but contributors can retrigger tests by closing and repoening them.

File: salt/pillar/consul_pillar.py (hunk 3 above)
Comment (devopsbrett) [inline salt/pillar/consul_pillar.py]: This has been done inside commit 6ca1dbd

**Pull request merged**